pkg/api/azenvtypes.go

```diff
@@ -93,9 +93,9 @@ var (
 		EtcdDownloadURLBase:              "mcr.microsoft.com/oss/etcd-io/",
 		KubeBinariesSASURLBase:           "https://acs-mirror.azureedge.net/kubernetes/",
 		WindowsTelemetryGUID:             "fb801154-36b9-41bc-89c2-f4d4f05472b0",
-		CNIPluginsDownloadURL:            "https://acs-mirror.azureedge.net/cni/cni-plugins-amd64-" + CNIPluginVer + ".tgz",
-		VnetCNILinuxPluginsDownloadURL:   "https://acs-mirror.azureedge.net/cni/azure-vnet-cni-linux-amd64-" + AzureCniPluginVerLinux + ".tgz",
-		VnetCNIWindowsPluginsDownloadURL: "https://acs-mirror.azureedge.net/cni/azure-vnet-cni-windows-amd64-" + AzureCniPluginVerWindows + ".zip",
+		CNIPluginsDownloadURL:            "https://acs-mirror.azureedge.net/cni-plugins/" + CNIPluginVer + "/binaries/cni-plugins-linux-amd64-" + CNIPluginVer + ".tgz",
+		VnetCNILinuxPluginsDownloadURL:   "https://acs-mirror.azureedge.net/azure-cni/" + AzureCniPluginVerLinux + "/binaries/azure-vnet-cni-linux-amd64-" + AzureCniPluginVerLinux + ".tgz",
```

Comment:
The change of the CNI url format  requires change to the install command in parts/k8s/cloud-init/artifacts/cse_install.sh, where it tries to get the file name. It should be something like "CNI_TGZ_TMP=${CNI_PLUGINS_URL##*/}"  and CNI_TGZ_TMP=${VNET_CNI_PLUGINS_URL##*/}

This change was not in aks-engine at the time of 0.47.0 vendoring.

After that we need to do "make generate" to update the templates_generated.go. 

If this sounds too much , I can do that for you.

downloadCNI() {
    mkdir -p $CNI_DOWNLOADS_DIR
    CNI_TGZ_TMP=$(echo ${CNI_PLUGINS_URL} | **cut -d "/" -f 5)**
    retrycmd_get_tarball 120 5 "$CNI_DOWNLOADS_DIR/${CNI_TGZ_TMP}" ${CNI_PLUGINS_URL} || exit $ERR_CNI_DOWNLOAD_TIMEOUT
}

downloadAzureCNI() {
    mkdir -p $CNI_DOWNLOADS_DIR
    CNI_TGZ_TMP=$(echo ${VNET_CNI_PLUGINS_URL} | **cut -d "/" -f 5**)
    retrycmd_get_tarball 120 5 "$CNI_DOWNLOADS_DIR/${CNI_TGZ_TMP}" ${VNET_CNI_PLUGINS_URL} || exit $ERR_CNI_DOWNLOAD_TIMEOUT
}

Reply:
Think I go this. Only changed for azure CNI.